### PR TITLE
changing wellen to a cargo workspace

### PR DESCRIPTION
**Pull request by 1024bees**
Hey, we talked at latch-up -- I'd like to add python bindings to wellen.

To do this practically, wellen needs to be structured as a cargo workspace, so the python bindings and the rust library can be decoupled -- took a non-invasive, first stab at this in this commit.

### CODE REVIEW

**Review by ekiwi — CHANGES_REQUESTED** [review]
This looks generally good. Thank you very much!

I _do_ like to keep my directory hierarchy as flat as possible. Could we get rid of the `crates` directory and put the new `wellen` folder directly into the repo root?

**Comment by 1024bees** [comment]
sgtm, i've moved it up to repo root

**Review by ekiwi — APPROVED** [review]
Thanks! I will merge this as soon as the CI passes.

**Comment by ekiwi** [comment]
@1024bees : Please have a look at the CI failures and try to address them. I think this is mostly about some paths that need to be fixed.

Also, please try to address these warnings:
```
warning: /home/runner/work/wellen/wellen/wellen/Cargo.toml: unused manifest key: package.default-members
warning: /home/runner/work/wellen/wellen/Cargo.toml: unused manifest key: workspace.package.default-members
```

**Comment by 1024bees** [comment]
woops, sorry about that, fixed the warnings and CI should be working. 

With the way workspaces have been implemented (virtual workspaces would be bette), for examples to live in root, each example needs it's own `Cargo.toml` manifest file -- more context provided at https://github.com/rust-lang/cargo/issues/7467. An example of this structure is here: https://github.com/tracel-ai/burn/tree/main/examples

I can do this if you'd prefer -- it might be the better long term structure for examples, but I'm indifferent to it right now. To keep friction low for this MR, examples have been moved into the `wellen` crate and `inputs/` have been symlinked from the workspace root. running examples should be exactly the same as before this MR, if you cd into the `wellen` crate.

File: wellen/inputs [hunk 1]
```diff
@@ -0,0 +1 @@
+../inputs
```
Comment (ekiwi) [inline wellen/inputs:1]: @1024bees Could you please remove the symlink and change the path in the CI settings instead?
Otherwise, this is good to go.

Comment (1024bees) [reply wellen/inputs]: done with https://github.com/ekiwi/wellen/pull/17/commits/50390d78828e04fecf8e4998d612498e66bf5fbd, thanks for all your patience :)

**Comment by 1024bees** [comment]
Not yet, I've been a little distracted in life, but I'd like to this and this is good motivation :). I'll look to have a draft up by ~Sunday

**Comment by ekiwi** [comment]
> Not yet, I've been a little distracted in life, but I'd like to this and this is good motivation :). I'll look to have a draft up by ~Sunday

That would be exciting. It is nice to know that you are still working on it.

**Pull request merged**